src/vs/base/browser/ui/splitview/paneview.ts

```diff
@@ -286,7 +287,7 @@ export abstract class Pane extends Disposable implements IView {
 	}
 
 	protected abstract renderHeader(container: HTMLElement): void;
-	protected abstract renderBody(container: HTMLElement): void;
+	protected abstract renderBody(container: HTMLElement, rootContainer: HTMLElement): void;
```

Comment:
This makes me itchy. 🙈

Reply:
I think this is a hack, but I also can't propose a good solution. Nevertheless, my rule for hacks is: keep them as contained as possible. In that mindset, couldn't we simply navigate up the parent DOM chain in `layoutWebview()`, find the element we assume to be the _root element_ and use that? The hack would not spread to other files and would also not change component API.

Reply:
Ok, I think that should be possible. As you note, it will require touching less code but could break if the parent dom structure changes (such as if an element's class is renamed)

I originally tried using intersection observers for this but couldn't get them working as I needed

Reply:
>  intersection observers

TIL!